lib/FolderRow/FolderRow.js

```diff
@@ -1,37 +1,71 @@
-import React from 'react';
+import React, { Fragment, useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
+import Button from '../Button';
+import Icon from '../Icon';
 
-const FolderRow = ({ children, actions, metaText, open, ...props }) => {
-  const classNames = cx('folder-row', {
-    'folder-row--open': open
+const FolderRow = ({
+  children,
+  name,
+  actions,
+  metaText,
+  open,
+  className,
+  showToggle,
+  ...props
+}) => {
+  const [toggle, setToggle] = useState(open);
+
+  useEffect(() => {
+    setToggle(open);
+  }, [open]);
+
+  const classNames = cx(`folder-row ${className}`, {
+    'folder-row--open': toggle
   });
 
   return (
-    <div className={classNames} {...props}>
-      <div className="folder-row__name">
-        <div className="text-overflow-ellipsis">{children}</div>
-      </div>
+    <Fragment>
+      <div className={classNames} {...props}>
+        <div className="folder-row__name">
+          <div className="text-overflow-ellipsis">{name}</div>
+        </div>
 
-      <div className="folder-row__meta">
-        {actions && <div className="folder-row__actions">{actions}</div>}
-        {metaText && <div className="folder-row__meta-text">{metaText}</div>}
+        <div className="folder-row__meta">
+          {actions && <div className="folder-row__actions">{actions}</div>}
+          {metaText && <div className="folder-row__meta-text">{metaText}</div>}
+          {showToggle && (
+            <div className="folder-row__toggle h-margin-left-half">
+              <Button types={['icon-only']} onClick={() => setToggle(!toggle)}>
+                <Icon name={toggle ? 'caretUp' : 'caret'} />
```

Comment:
Not for now, but I feel like this could be useful to split out into a little helper, its a pattern thats pretty common